src/main/java/com/minecolonies/client/gui/AbstractWindowBuilding.java

```diff
@@ -0,0 +1,105 @@
+{
+    private static final String BUTTON_BUILD        = "build";
+    private static final String BUTTON_REPAIR       = "repair";
+    private static final String BUTTON_INVENTORY    = "inventory";
+    private static final String LABEL_BUILDING_NAME = "name";
+
+    /**
+     * Type B is a class that extends {@link AbstractBuildingWorker.View}.
+     */
+    protected final B building;
+
+    /**
+     * Constructor for the windows that are associated with buildings.
+     *
+     * @param building      Class extending {@link AbstractBuildingHut.View}.
+     * @param resource      Resource location string.
+     */
+    public AbstractWindowBuilding(final B building, final String resource)
+    {
+        super(resource);
+
+        this.building = building;
+        registerButton(BUTTON_BUILD, this::buildClicked);
+        registerButton(BUTTON_REPAIR, this::repairClicked);
+        registerButton(BUTTON_INVENTORY, this::inventoryClicked);
+    }
+
+    /**
+     * Action when build button is clicked
+     *
+     * @param ignored   Parameter is ignored, since some actions require a button.
+     *                  This method does not
+     */
+    private void buildClicked(Button ignored)
+    {
+        MineColonies.getNetwork().sendToServer(new BuildRequestMessage(building, BuildRequestMessage.BUILD));
+    }
+
+    /**
+     * Action when repair button is clicked
+     *
+     * @param ignored   Parameter is ignored, since some actions require a button.
+     *                  This method does not
+     */
+    private void repairClicked(Button ignored)
+    {
+        MineColonies.getNetwork().sendToServer(new BuildRequestMessage(building, BuildRequestMessage.REPAIR));
+    }
+
+    /**
+     * Action when a button opening an inventory is clicked
+     *
+     * @param ignored   Parameter is ignored, since some actions require a button.
+     *                  This method does not
+     */
+    private void inventoryClicked(Button ignored)
+    {
+        MineColonies.getNetwork().sendToServer(new OpenInventoryMessage(building));
+    }
+
+    /**
+     * Called when the Window is displayed.
+     */
+    @Override
+    public void onOpened()
+    {
+        findPaneOfTypeByID(LABEL_BUILDING_NAME, Label.class).setLabelText(LanguageHandler.getString(getBuildingName()));
+
+        if (building.getBuildingLevel() == 0)
+        {
+            findPaneOfTypeByID(BUTTON_BUILD, Button.class).setLabel(LanguageHandler.getString("com.minecolonies.gui.workerHuts.build"));
+            findPaneByID(BUTTON_REPAIR).disable();
+        }
+        else if (building.isBuildingMaxLevel())
+        {
+            Button button = findPaneOfTypeByID(BUTTON_BUILD, Button.class);
+            button.setLabel(LanguageHandler.getString("com.minecolonies.gui.workerHuts.upgradeUnavailable"));
+            button.disable();
+        }
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)